docs/sql-migration-guide-upgrade.md

```diff
@@ -147,6 +147,8 @@ license: |
 
   - Since Spark 3.0, if files or subdirectories disappear during recursive directory listing (i.e. they appear in an intermediate listing but then cannot be read or listed during later phases of the recursive directory listing, due to either concurrent file deletions or object store consistency issues) then the listing will fail with an exception unless `spark.sql.files.ignoreMissingFiles` is `true` (default `false`). In previous versions, these missing files or subdirectories would be ignored. Note that this change of behavior only applies during initial table file listing (or during `REFRESH TABLE`), not during query execution: the net change is that `spark.sql.files.ignoreMissingFiles` is now obeyed during table file listing / query planning, not only at query execution time.
 
+  - Since Spark 3.0, substitution order of nested WITH clauses is changed and an inner CTE definition takes precedence over an outer. The previous behaviour can be restored by setting `spark.sql.legacy.outerCTEHasPriority` to `true`.
```

Comment:
cc @gatorsmile

Reply:
could you add a simple example query here? (I remember this suggested in https://issues.apache.org/jira/browse/SPARK-24175...)

Reply:
Sure, good idea, I've added it.